DataFormats/METReco/src/HcalNoiseHPD.cc

```diff
@@ -110,28 +110,28 @@ const edm::RefVector<HBHERecHitCollection> HcalNoiseHPD::recHits(void) const
   return rechits_;
 }
   
-float HcalNoiseHPD::recHitEnergy(float threshold) const
+float HcalNoiseHPD::recHitEnergy(const float threshold) const
 {
-  float total=0.0;
+  double total=0.0;
   for(edm::RefVector<HBHERecHitCollection>::const_iterator it=rechits_.begin(); it!=rechits_.end(); ++it) {
-    float energy=(*it)->energy();
+    const float energy=(*it)->eraw();
     if(energy>=threshold) total+=energy;
   }
   return total;
 }
 
-float HcalNoiseHPD::recHitEnergyFailR45(float threshold) const
+float HcalNoiseHPD::recHitEnergyFailR45(const float threshold) const
```

Comment:
It is kind of in the spirit "Use const whenever possible" -- see item 21 in Scott Meyers' "Effective C++" book.